components/main-chef-wrapper/lib/version.go

```diff
@@ -0,0 +1,142 @@
+	manifestMap = manifestHash()
+}
+func Version(){
+	if omnibusInstall() == true {
+		showVersionViaVersionManifest()
+	} else {
+		fmt.Fprintln(os.Stderr, "ERROR:", dist.WorkstationProduct, "has not been installed via the platform-specific package provided by", dist.DistributorName, "Version information is not available.")
+
+
+	}
+}
+
+func showVersionViaVersionManifest()  {
+	fmt.Printf("%v version: %v", dist.WorkstationProduct, componentVersion("build_version")  )
+	productMap := map[string]string{
+		dist.ClientProduct: dist.CLIWrapperExec,
+		dist.InspecProduct: dist.InspecCli,
+		dist.CliProduct: dist.CliGem,
+		dist.HabProduct: dist.HabSoftwareName,
+		"Test Kitchen": "test-kitchen",
+		"Cookstyle": "cookstyle",
+	}
+	for prodName, component := range productMap {
+		fmt.Printf("\n%v version: %v", prodName, componentVersion(component)  )
+	}
+	fmt.Printf("\n")
+}
+func componentVersion(component string) string  {
+	v, ok := gemManifestMap[component]
+	if ok {
+		stringifyVal := v.([]interface{})[0]
+		return stringifyVal.(string)
+	} else if v, ok := manifestMap[component]; ok {
+		return v.(string)
+	} else {
+		success, _ := Dig(manifestMap, "software", component, "locked_version")
+		if success == nil {
+			return "unknown"
+		}else {
+			return success.(string)
+		}
+	}
+}
+func gemManifestHash() map[string]interface{} {
+	filepath := path.Join(omnibusRoot(),"gem-version-manifest.json")
+	jsonFile, err := os.Open(filepath)
+	if err != nil {
+		fmt.Println(err)
+	}
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+	var gemManifestHash map[string]interface{}
+	json.Unmarshal([]byte(byteValue), &gemManifestHash)
+	defer jsonFile.Close()
+	return gemManifestHash
+}
+func manifestHash() map[string]interface{}  {
+	filepath := path.Join(omnibusRoot(),"version-manifest.json")
+	jsonFile, err := os.Open(filepath)
+	if err != nil {
+		fmt.Println(err)
+	}
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+	var manifestHash map[string]interface{}
+	json.Unmarshal([]byte(byteValue), &manifestHash)
+	defer jsonFile.Close()
+	return manifestHash
+}
+
+func omnibusInstall() bool {
+	//# We also check if the location we're running from (omnibus_root is relative to currently-running ruby)
+	//# includes the version manifest that omnibus packages ship with. If it doesn't, then we're running locally
+	//# or out of a gem - so not as an 'omnibus install'
+	ExpectedOmnibusRoot := ExpectedOmnibusRoot()
+	if _, err := os.Stat(ExpectedOmnibusRoot); err == nil {
+		if _, err = os.Stat(path.Join(ExpectedOmnibusRoot,"version-manifest.json")); err == nil {
+			return true
+		} else {
+			return false
+		}
+	} else {
+		return false
+	}
+}
+
+
+func omnibusRoot()string  {
+	omnibusroot, err := filepath.Abs(path.Join(ExpectedOmnibusRoot()))
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "ERROR:", dist.WorkstationProduct, "has not been installed via the platform-specific package provided by", dist.DistributorName, "Version information is not available.")
```

Comment:
We should probably stop execution here, since returning the empry/nil string will cause further issues in the caller; or we can alternatively  return the error here and exit on error in the caller.